src/js/components/style-item.jsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { getImagePos } from '../lib/utils.js'
+import image from '../lib/style-image.json'
+import { useViewport } from '../store/use-viewport'
+
+const getStyleItemDisplayName = (item) => {
+  if (item.displayName) {
+    return item.displayName
+  }
+  return {
+    default: 'Default',
+    dark: 'Dark',
+    aerial: 'Aerial',
+    deuteranopia: 'Green-red enhanced',
+    tritanopia: 'Blue-yellow enhanced'
+  }[item.name] || item.name
+}
+
+const StyleItem = ({ key, item, currentStyleName, ref }) => {
+  const viewportDispatch = useViewport().dispatch
+  const handleStyleClick = e => {
+    viewportDispatch({ type: 'SET_STYLE', payload: { style: e.currentTarget.value } })
+  }
+
```

Comment:
When reviewing: 

The only real change here is the new getStyleItemDisplayName method here.

All the rest was hoisted from the Parent component and moved to a new one, almost entirely unchanged.